aws-sensu.md

```diff
@@ -0,0 +1,32 @@
+## Notes from monitoring AWS from Sensu Summit
+
+Grafana has a CloudWatch plugin you can use to access the data directly rather than pulling to TSDB
+data lags and by default is 5min aggregation.  Not generally very useful for granularity.
+They’ve recently added some dashboarding but it’s not very good compared to Grafana
+
+### Good practices/checks to implement:
+  * Eric wrote a check all target groups in the region for ALB - https://github.com/sensu-plugins/sensu-plugins-aws/blob/master/bin/check-alb-target-group-health.rb
+  * Check for instances that are impaired - https://github.com/sensu-plugins/sensu-plugins-aws/blob/master/bin/check-instance-health.rb
+  * Reconfigure autoscale groups via handlers based on metrics - https://github.com/sensu-plugins/sensu-plugins-aws/blob/master/bin/handler-scale-asg-up.rb
+  * gather info for billing purposes using billing data in cloudwatch
+  * check for detached volumes/untagged instances/etc - https://github.com/sensu-plugins/sensu-plugins-aws/blob/master/bin/check-ec2-cpu_balance.rb
+  * Checking for CPU credits is a good practice to avoid unexpected throttling affecting the application when something has gone off the rails.  Cloudwatch check is helpful for this as it can be exposed as a metric
+
+### Building checks:
+  * use SDKs but be aware of what version
+  * https://github.com/sensu-plugins/sensu-plugins-aws
+  * Build with config management tool so devs/ops can add a new ALB/EC2/whatever to a list and then iterate that list to build checks as needed
```

Comment:
I am not sure I follow, are you suggesting creating multiple instantiations of the checks? If so this is a very expensive way of doing it I would suggest running through all the checks possible and make sure that they can all take a comma separated list of resources so it can reduce the number of the `aws-sdk` is needed to be loaded which reduces load significantly. I did this with my SQS checks and went from 50 -> 3 by grouping them together (plugin needed to be re-written) which reduced overall CPU usage from being pegged 100% all the time to an average of 60-70% load.